store/views.py

```diff
@@ -77,6 +93,54 @@ def loanBookView(request):
 @csrf_exempt
 @login_required
 def returnBookView(request):
-    pass
+    book_id = request.POST.get("bid")
+    book = BookCopy.objects.get(pk=book_id)
+    if(book):
+      message ="success"
+      book.status = True
+      book.borrower = None
+      book.borrow_date = None
+      book.save()
+    else :
+      message = "failure" 
+    response_data = {
+        'message': message,
+    }
+    return JsonResponse(response_data)
 
 
+@csrf_exempt
+@login_required
+def userBookRating(request):
+    response_data = {
+        'message': "failure",
+        'rating' : 0 
+    }
+    prating = '0'
+    userid=request.user.id
+    data = request.POST
+    rating =data.get('rating')
+    bookid = data.get('bid')
+    print(rating)
+    if BookRating.objects.filter(user_id=userid).filter(book_id=bookid).count() == 1:
+        prating = BookRating.objects.filter(user_id=userid).filter(book_id=bookid).first().rating
+        print('rating')
+    if prating == '0':
+        rate = BookRating.objects.create(book_id=bookid,user_id=userid,rating=rating)
+        response_data['message'] = "success"
+    else :
+        rate=BookRating.objects.filter(user_id=userid).filter(book_id=bookid).first()
+        rate.rating = rating
+        rate.save()
+        response_data['message'] = "success"
+    book=Book.objects.get(id=bookid)
+    sum=0
+    l= BookRating.objects.filter(book_id=bookid).count()
+    for i in range(l):
+        rate=BookRating.objects.filter(book_id=bookid)[i]
+        sum = sum + int(rate.rating)
```

Comment:
This is not a good way to call ORM queries. This will run two queries on the database.
You could have done it this way:
```python
for rate in Rating.objects.filter(book_id=bookid):
    sum = sum + int(rate.rating)
```